crypto/pkcs12/p12_init.c

```diff
@@ -25,6 +25,17 @@ PKCS12 *PKCS12_init(int mode)
     if (!ASN1_INTEGER_set(pkcs12->version, 3))
         goto err;
     pkcs12->authsafes->type = OBJ_nid2obj(mode);
+    pkcs12->authsafes->ctx.libctx = ctx;
```

Comment:
maybe it should have a helper for this in pk7_asn1.c (maybe a separate PR since this one has been going for a while).
    pcks7_set0_libctx(pkcs12->authsafes, libctx, propq);

Reply:
Looks like you already have these methods - they are just static currently. So expose them internally.